### PR TITLE
Instant Search: setup preact-testing-library

### PR DESCRIPTION
**Pull request by bluefuton**
Sets up automated testing for our Instant Search Preact app.

https://github.com/testing-library/preact-testing-library#docs

#### Changes proposed in this Pull Request:

* No functional changes.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:

Try `yarn test-search` and `yarn test-extensions`, and ensure that the tests pass.

#### Proposed changelog entry for your changes:

* No changelog required. Merges into `instant-search-master`.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const path = require( 'path' );
 
 module.exports = {
 	preset: '@automattic/calypso-build',
-	roots: [ '<rootDir>/extensions/' ],
+	roots: [ '<rootDir>/extensions/', '<rootDir>/modules/search/instant-search' ],
```
Comment (bluefuton) [inline jest.config.js]: Possibly should split this out - this ends up being run by `yarn test-extensions`.

**Comment by bluefuton** [comment]
@jsnmoon can you see anything missing in this sample test?

https://github.com/Automattic/jetpack/pull/13950/files#diff-a73a4a62b62e24a82a61dd2d585e4fd2

Seems to just render an empty div, whatever component I provide it :/

```
 <body>
        <div />
      </body>
```

File: jest.config.js (hunk 1 above)
Comment (bluefuton) [reply jest.config.js]: Created `yarn test-search`.

File: package.json [hunk 2]
```diff
@@ -166,6 +167,9 @@
 	},
 	"devDependencies": {
 		"@slack/web-api": "5.3.0",
+		"@slack/web-api": "5.2.0",
```
Comment (bluefuton) [inline package.json]: Shouldn't be trying to downgrade this! I'll fix.

**Comment by bluefuton** [comment]
Also need to remove the package-lock.json (thanks @jsnmoon)

**Review by jsnmoon — APPROVED** [review]
Nice, it's finally working! 🚢

**Pull request merged**